### PR TITLE
Fix iOS 8 bug with assert fail on infinity CGFloat

### PR DESCRIPTION
**Pull request by requilence**
iOS 8 produce 'inf' CGFloat instead of CGFLOAT_MAX on some CGPath (some bezier paths). But fails when try to assert this CGFloat as a valid float.
"Assertion failed: (CGFloatIsValid(x) && CGFloatIsValid(y)), function void CGPathMoveToPoint(CGMutablePathRef, const CGAffineTransform *, CGFloat, CGFloat), file Paths/CGPath.cc, line 254."

Final iOS 8.1 SDK and XCode 6.1 released but this bug remains. So there is a need for workaround.
This fix replace inf with CGFLOAT_MAX and -inf to -CGFLOAT_MAX in the returned coordinates by quartz

### CODE REVIEW

File: Source/QuartzCore additions/CGPathAdditions.m [hunk 1]
```diff
@@ -15,14 +15,31 @@
...
 
+CGFloat fixInfinity(CGFloat inputFloat){
+    if(inputFloat>CGFLOAT_MAX) inputFloat=CGFLOAT_MAX;
+    if(inputFloat<(-1)*CGFLOAT_MAX) inputFloat=(-1)*CGFLOAT_MAX;
+    return inputFloat;
+}
+
+CGPoint *fixPointInfinity(CGPoint *inputFloat){
+    int i;
+    for (i = 0; i < sizeof(inputFloat); ++i)
```
Comment (MaddTheSane) [inline Source/QuartzCore additions/CGPathAdditions.m]: This doesn't look right…

File: Source/QuartzCore additions/CGPathAdditions.m [hunk 2]
```diff
@@ -15,14 +15,31 @@
 	CGFloat offY;
 } PathInfo;
 
+CGFloat fixInfinity(CGFloat inputFloat){
+    if(inputFloat>CGFLOAT_MAX) inputFloat=CGFLOAT_MAX;
+    if(inputFloat<(-1)*CGFLOAT_MAX) inputFloat=(-1)*CGFLOAT_MAX;
+    return inputFloat;
+}
+
+CGPoint *fixPointInfinity(CGPoint *inputFloat){
```
Comment (MaddTheSane) [inline Source/QuartzCore additions/CGPathAdditions.m]: If this is iterating over a C array, you probably want to pass the size of the array.

**Comment by requilence** [comment]
@MaddTheSane Thank you. Forget about that thing. Fixed by switching of CGPathElementType.

**Comment by adamgit** [comment]
Ouch, so that's what was going wrong. Good catch!

**Comment by adamgit** [comment]
For bonus points, do you have a small SVG which shows this problem, and you can share / make public domain that we can add to the Demo project?

(Apple might change this again in future, it would be great to have a unit test to help in future)

**Pull request merged**